dialogue-target/src/test/java/com/palantir/dialogue/RequestTest.java

```diff
@@ -88,4 +92,38 @@ void from_method_query_param_mutation() {
 
         System.out.println(request2.queryParams());
     }
+
+    @Test
+    void copy_request_with_existing_params() {
+        Request oldRequest = Request.builder()
+                .putQueryParams("query1", "queryVal1")
+                .putHeaderParams("header1", "headerVal1")
+                .putPathParams("path1", "pathVal1")
+                .build();
+
+        Request newRequest = Request.builder()
+                .from(oldRequest)
+                .putQueryParams("query2", "queryVal2")
+                .putHeaderParams("header2", "headerVal2")
+                .putPathParams("path2", "pathVal2")
+                .build();
+
+        // make sure new query/header/path parameters were added in addition to whatever was in `oldRequest`
+        assertThat(newRequest)
+                .extracting(Request::queryParams)
+                .extracting(Multimap::entries, collection(Map.Entry.class))
+                .containsExactlyInAnyOrder(
+                        MapEntry.entry("query1", "queryVal1"), MapEntry.entry("query2", "queryVal2"));
```

Comment:
would be good to also test unique query param in each of old & new requests as well as duplicate query param names (since they're a `ListMultimap`)